app/src/main/java/org/simple/clinic/sync/SyncScheduler.kt

```diff
@@ -22,34 +22,62 @@ class SyncScheduler @Inject constructor(
     return Observable.fromIterable(syncs)
         .flatMapSingle { it.syncConfig() }
         .distinct { it.syncGroup }
-        .flatMapSingle(this::createWorkRequest)
+        .map { config -> createWorkRequest(config) to config.syncGroup.name }
         .toList()
-        .flatMapCompletable { workRequests ->
-          Completable.fromAction {
-            workManager.cancelAllWorkByTag(SyncWorker.TAG)
-            workManager.enqueue(workRequests)
-          }
-        }
+        .doOnSuccess { cancelPreviouslyScheduledPeriodicWork() }
+        .flatMapCompletable(this::scheduleWorkRequests)
+  }
+
+  /*
+   * This is meant to cancel the old periodic work that
+   * was scheduled and persisted before we moved to the
+   * "unique" work system.
+   * TODO 2019-09-02: Remove once the unique work feature has been deployed to enough devices
+   **/
+  private fun cancelPreviouslyScheduledPeriodicWork() {
+    workManager.cancelAllWorkByTag("patient-sync")
```

Comment:
Hard coded tag?

Reply:
Yes, this is temporary and will be removed once enough users migrate to the newer build.

Reply:
This is an implementation detail because the tag is persisted to WorkManager's internal work database, so we need to keep this string somewhere. However, we do not want to make this available to the rest of the codebase.